src/shapes.ts

```diff
@@ -0,0 +1,270 @@
+        const step_eta = 360 / resolution;
+        const step_omega = 180 / resolution;
+        const r = this.radius;
+
+        const indices = [];
+        const vertices = [];
+        const normals = [];
+
+        let eta, omega;
+        let eta_rad, omega_rad;
+        let index = 0;
+
+        for (
+            eta = -90;
+            eta <= 90;
+            eta += step_eta //I could do the cycle in rad but I can have problems in condition of termination of the cycle (float value)
+        ) {
+            eta_rad = (eta * Math.PI) / 180.0;
+
+            for (omega = -180; omega <= 180; omega += step_omega, index += 3) {
+                omega_rad = (omega * Math.PI) / 180.0;
+
+                const relativeVertex: XYZ = {
+                    x: r * Math.cos(eta_rad) * Math.cos(omega_rad),
+                    y: r * Math.cos(eta_rad) * Math.sin(omega_rad),
+                    z: r * Math.sin(eta_rad),
+                };
+
+                const newVertex: XYZ = {
+                    x: relativeVertex.x + this.position[0],
+                    y: relativeVertex.y + this.position[1],
+                    z: relativeVertex.z + this.position[2],
+                };
+                vertices.push(newVertex);
+
+                const newNormal: XYZ = {
+                    x: relativeVertex.x / Math.pow(r, 2.0),
+                    y: relativeVertex.y / Math.pow(r, 2.0),
+                    z: relativeVertex.z / Math.pow(r, 2.0),
+                };
+                normals.push(newNormal);
+            }
+        }
+
+        //Now triangulation. I believe that it needs a counterclockwise specification of vertices (else I have back-face)
+        //each triangle is specified with 3 indices (gl.TRIANGLES).
+
+        const num_samples_omega = 360 / step_omega; //Actually there is another sample that coincides with the first sample.
+        const num_samples_eta = 180 / step_eta;
+
+        let i, j;
+        for (i = 0, index = 0; i < num_samples_eta; i++) {
+            for (j = 0; j < num_samples_omega; j++, index += 6) {
+                indices[index] = i * num_samples_omega + i + j;
+                indices[index + 1] = (i + 1) * num_samples_omega + 2 + i + j;
+                indices[index + 2] = (i + 1) * num_samples_omega + 1 + i + j;
+
+                indices[index + 3] = i * num_samples_omega + i + j;
+                indices[index + 4] = i * num_samples_omega + 1 + i + j;
+                indices[index + 5] = (i + 1) * num_samples_omega + 2 + i + j;
+            }
+        }
+        return {
+            vertexArr: vertices,
+            normalArr: normals,
+            faceArr: indices,
+            color: color,
+        };
+    }
+}
+
+export class Ellipsoid extends Shape {
+    public semiaxes: [number, number, number];
+
+    constructor(
+        semiaxes: [number, number, number],
+        pos: [number, number, number] = [0, 0, 0],
+        orientation: [number, number, number, number] = [1, 0, 0, 0]
+    ) {
```

Comment:
```suggestion
        constructor(position: [number, number, number], data: EllipsoidData) {
```

Reply:
I had to revert this. Issues with ShapeData = EllipsoidData | SphereData

Reply:
Done!